File: typescript/src/api/resources/evaluationComparisons/client/requests/ComparisonInput.ts

import * as Sikaru from "../../../../index.js";

/**
 * @example
 *     {
 *         cases: [{
 *                 baseline: {
 *                     accountId: "accountId",
 *                     conversationId: "conversationId",
 *                     messageId: "messageId"
 *                 },
 *                 candidate: {
 *                     accountId: "accountId",
 *                     conversationId: "conversationId",
 *                     messageId: "messageId"
 *                 },
 *                 caseId: "caseId"
 *             }],
 *         evaluator: "evaluator",
 *         id: "id",
 *         revision: "revision",
 *         rubric: "rubric"
 *     }
 */
export interface ComparisonInput {
    cases: Sikaru.CaseInput[];
    evaluator: string;
    id: string;
    revision: string;
    rubric: string;
}

File: ruby/lib/Sikaru/evaluation_comparisons/types/comparison_input.rb
# frozen_string_literal: true

module Sikaru
  module EvaluationComparisons
    module Types
      class ComparisonInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :cases, -> { Internal::Types::Array[Sikaru::Types::CaseInput] }, optional: false, nullable: false

        field :evaluator, -> { String }, optional: false, nullable: false

        field :id, -> { String }, optional: false, nullable: false

        field :revision, -> { String }, optional: false, nullable: false

        field :rubric, -> { String }, optional: false, nullable: false
      end
    end
  end
end
